File: OS X/TPGradientOutlineView.m
//
//  TPGradientOutlineView.m
//  TwinPeaks
//
//  Created by marc hoffman on 12/16/09.
//  Copyright 2009 RemObjects Software. All rights reserved.
//

#import "TPGradientOutlineView.h"
#import "TPGradientShared.h"

@implementation TPGradientOutlineView

#include "TPGradientSharedCode.h"

- (NSImage *)dragImageForRowsWithIndexes:(NSIndexSet *)dragRows tableColumns:(NSArray *)tableColumns event:(NSEvent *)dragEvent offset:(NSPointPointer)dragImageOffset
{
	if ([self.delegate respondsToSelector:@selector(tableView:needsImageForDraggingRowsWithIndexes:)])
	{
		return [(id<TPGradientTableViewDelegate>)self.delegate tableView:self needsImageForDraggingRowsWithIndexes:dragRows];
	}
	return [super dragImageForRowsWithIndexes:dragRows 
								 tableColumns:tableColumns 
										event:dragEvent 
									   offset:dragImageOffset];
}

@end

File: OS X (C#)/TPGradientOutlineView.cs
using AppKit;

namespace RemObjects.TwinPeaks.OSX
{
	public class TPGradientOutlineView: NSOutlineView
	{
		public override NSImage dragImageForRowsWithIndexes(NSIndexSet dragRows) tableColumns(NSArray tableColumns) @event(NSEvent dragEvent) offset(NSPointPointer dragImageOffset)
		{
			if (this.@delegate.respondsToSelector(__selector(tableView:needsImageForDraggingRowsWithIndexes:)))
			{
				return (this.@delegate as ITPGradientTableViewDelegate).tableView(this) needsImageForDraggingRowsWithIndexes(dragRows);
			}
			return base.dragImageForRowsWithIndexes(dragRows) tableColumns(tableColumns) @event(dragEvent) offset(dragImageOffset);
		}
	}
}
